thaomonster/rancid-tomatillos
# Be/feature/api calls

### What does this PR do? 
This branch uses the `fetch` api to rely on server side data instead of the dummy data in our `src` directory. The `data` directory has since been deleted. This is very much an MVP and we still need some improvements like a loading screen, and better error messages.

### Is this a feature or Fix/Refactor?  
Feature.

### Problems Encountered & Resulting Solutions  
I had some issues getting error handling to work. Right now the code that handles errors is a little sloppy, and I'm not exactly sure why we need the conditionals before we parse responses in order for `.catch` to run. Open to feedback and I'll ask my mentor for help understanding it and what we can do better early next week.

### Where should the reviewer start?
First just spin up the app and make sure everything still works. Then look at the fetch requests in `App.js` and follow those new methods as they get passed down to child components. Most of the logic is in `App.js` but some other components have changed slightly too. 

### How should this be tested?
Same as before, spin up the app and make sure it works. Then maybe mess with the request urls to get bad responses and test out the error handling. Not much has changed on this branch function wise so no news is good news. 

### Steps To Take
We need a few things to improve the app at this stage: 
-styling, and refactored CSS
-better error messages and a loading screen.
-a look at the app's accessibility
